### ISSUE TITLE
closeness sort does not work as expected if no sortParam provided

### ISSUE THREAD

**Issue by romainneutron** [alchemy-fr/GeonamesServer#44]
When sorting by closeness, sort should be done against the client IP address if no sortParam is provided.

This is actually not the case, here are the result for `paris`, from paris, with and without sort params : 

without : 

parameters are : 

```
array(
            'sort' => 'closeness',
            'name' => $place,
            'limit' => 1,
);
```

result is : 

```
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 17 Jun 2013 22:34:32 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 523
Connection: keep-alive
X-Powered-By: Express
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: X-Requested-With, origin, content-type, accept
Access-Control-Allow-Methods: OPTIONS, GET, POST
X-Geonames-sortby: population

{
  "geonames": {
    "totalResultsCount": "1",
    "geoname": [
      {
        "geonameid": "6837753",
        "title": "Paris",
        "country": "Indonesia",
        "match": {
          "title": "Paris",
          "country": "Indonesia"
        },
        "population": "0",
        "location": {
          "latitude": "-7.52",
          "longitude": "112.56"
        },
        "names": [
          "paris",
          "paris"
        ],
        "region": "East Java",
        "title_alt": "paris"
      }
    ]
  }
}
```

with :

parameters are : 

```
array(
            'sort' => 'closeness',
            'name' => $place,
            'limit' => 1,
            'sortParams'=>array(
                'ip' => '5.49.69.114'
            )
)
```

result is 

```
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 17 Jun 2013 22:36:21 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 503
Connection: keep-alive
X-Powered-By: Express
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: X-Requested-With, origin, content-type, accept
Access-Control-Allow-Methods: OPTIONS, GET, POST
X-Geonames-sortby: closeness

{
  "geonames": {
    "totalResultsCount": "1",
    "geoname": [
      {
        "geonameid": "2988496",
        "title": "Parisot",
        "country": "France",
        "match": {
          "title": "Parisot",
          "country": "France"
        },
        "population": "0",
        "location": {
          "latitude": "44.83",
          "longitude": "0.9"
        },
        "names": [
          "parisot"
        ],
        "region": "Aquitaine",
        "title_alt": "parisot"
      }
    ]
  }
}
```

You can note that the accuracy is not good

**Comment by romainneutron** [comment]
See new IP detection here https://groups.google.com/forum/#!msg/express-js/tqiKZKew0oI/kn27alYiO8wJ

**Comment by romainneutron** [comment]
Please note that population of 0 are returned when sort by population is activated

**Comment by nlegoff** [comment]
Fixed in #50
